### ISSUE TITLE
Executing the FlashLoanArbitrage gives a Gas estimation error

### ISSUE THREAD

**Issue by intervations** [jspruance/aave-flash-loan-tutorial#3]
Thanks for the great video walk-through tutorial on YouTube for this.
I am trying the run the arbitrage function with dex contract and always get the following error,

`Gas estimation errored with the following message (see below). The transaction execution will likely fail. Do you want to force sending?
Returned error: {"jsonrpc":"2.0","error":"execution reverted: Check the token allowance","id":2683750395878379}`

Obviously, if I choose to continue it fails like it says it will.
Is this a code problem or a problem with remix.ethereum.org?

I appreciate any advice given on this issue.

**Comment by BigGod12** [comment]
Hello, were you able to get it fixed, i`m facing same thing for some couples of days now and can not continue anymore, though i have all the necessary token funded for liquidity and even some etha sent to both dex.sol and the arbitrage contract, were you able to get it fixed please??

**Comment by intervations** [comment]
No. I stepped away from this as a project and am only just getting back to it.
I have had to rebuild my VM environment for this.

**Comment by leonidafanas** [comment]
Hello! Could You please provide an update on this? Was this error fixed? Once You will rebuild your VM environment - will You tell us what was a problem? Thanks! Just want to continue learning.

**Comment by intervations** [comment]
> Hello! Could You please provide an update on this? Was this error fixed? Once You will rebuild your VM environment - will You tell us what was a problem? Thanks! Just want to continue learning.

Hello. No this was never resolved. No-one who follows this repo seems to have an answer for the issue, not even the author has responded.

For now I have moved on and have forgotten about flashloans unless the question can be answered.

**Comment by leonidafanas** [comment]
My Arbitrage has finally worked. I think my issue was that I did not replaced 'import' IERC20 in Dex.sol to https://github. Etherscan has shown transactions stamps and balance on my FlashLoanArbitrage.sol is about 110$ just like in author's example. However, now, after my arbitrage could make it just once - when I try to make it work second time - it has that same issue with Gas Estimation Error. I tried to add some liquidity to Dex.sol and seems that both USD and DAI balances are more than enough, but still can's do it second time.

**Comment by testmbout** [comment]
I have the same question, but when i change the deploy address to "0x0Dc26cC61c1AD92D12804849C58EF90AA1323c33", this address I get from another issue,  then it worked well, I think it about the key of quetion
